keps/sig-api-machinery/1693-warnings/README.md

```diff
@@ -79,27 +79,27 @@ and manually sweeping for use of deprecated features.
 * Allow a cluster administrator to programatically identify particular clients using deprecated APIs
 
 ### Non-Goals
-
+ 
 While the proposed warning mechanism is generic enough to carry arbitrary warnings,
 the following items are out of scope for the first iteration of this feature:
 
-* Allowing extensions mechanisms like admission webhooks to contribute warnings
+* Allowing extension mechanisms like admission webhooks to contribute warnings
 * Surfacing warnings about other non-fatal problems
```

Comment:
This exclusion surprises me.  We would reject PRs adding this capability or you simply aren't adding it?

Reply:
That's a second design effort that includes the API between the admission webhooks and the API server. I'm in favor of it eventually, but not in this first iteration.

Reply:
I'm in favor of it eventually, but not in this first iteration. The main things I'd like to see considered are guidelines around volume and structure of messages

Reply:
> I'm in favor of it eventually, but not in this first iteration. The main things I'd like to see considered are guidelines around volume, structure of messages, and under what circumstances a warning would be produced (on creation? on update? on change of a field/field-value that deserves a warning?)

I agree that it doesn't need to be in this iteration, but I would like to see it on the criteria for graduation to GA.  From a plumbing perspective and a client experience perspective, this enhancement is about warnings.  Deprecation happens to be the first one we're doing, but you're creating WarningRecorder, using a warning standard, and honoring warnings in the client.

Reply:
moved to GA criteria